src/lib.rs

```diff
@@ -346,6 +402,54 @@ impl GenerateTasksServiceImpl {
             .filter(|t| t != &self.generating_task)
             .collect()
     }
+
+    /// Determine which distro the given sub-tasks should run on.
+    ///
+    /// By default, we won't specify a distro and they will just use the default for the build
+    /// variant. If they specify `use_large_distro` then we should instead use the large distro
+    /// configured for the build variant. If that is not defined, then throw an error unless
+    /// the build variant is configured to be ignored.
+    ///
+    /// # Arguments
+    ///
+    /// * `large_distro_name` - Name
+    fn determine_distro(
+        &self,
+        large_distro_name: &Option<String>,
+        generated_task: &dyn GeneratedSuite,
+        build_variant_name: &str,
+    ) -> Result<Option<String>> {
+        if generated_task.use_large_distro() {
+            if large_distro_name.is_some() {
+                return Ok(large_distro_name.clone());
+            }
+
+            if let Some(gen_task_config) = &self.gen_sub_tasks_config {
+                if gen_task_config.ignore_missing_large_distro(build_variant_name) {
+                    return Ok(None);
+                }
+            }
+
+            bail!(
+                r#"
```

Comment:
Can you move this to be in a constant somewhere?

Reply:
No. The problem is that this is not a static string. It is a formatted string. You can see the formatting with `{build_variant_name}`. Rust is able to extract values to use from environment, but the cost is that the string needs to be inlined. In order to be able to pull something like this into a constant, we would need to pull in a 3rd party templating library, which seems expensive for what it would get us.

Reply:
Ah, sorry, I missed the `{build_variant_name}` portion of this. Yes, pulling in a different library seems excessive, this is fine